### PR TITLE
reformat dynamic metadata emitted by Mongo proxy

### PR DESCRIPTION
**Pull request by rshriram**
*Description*: Emit metadata as `map<resource, list(operations>` so that it can be used in metadata matchers easily. The existing format (messages:list(structs)) is too hard to represent in metadata matchers.
*Risk Level*: LOW
*Testing*: Unit tests

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

cc @venilnoronha

### CODE REVIEW

File: test/extensions/filters/network/mongo_proxy/proxy_test.cc [hunk 1]
```diff
@@ -246,15 +244,10 @@ TEST_F(MongoProxyFilterTest, DynamicMetadata) {
...
-  auto& multiple_messages = metadata.fields().at("messages").list_value();
-  EXPECT_EQ("OP_QUERY",
-            multiple_messages.values(0).struct_value().fields().at("operation").string_value());
-  EXPECT_EQ("db1.test1",
-            multiple_messages.values(0).struct_value().fields().at("resource").string_value());
-  EXPECT_EQ("OP_INSERT",
-            multiple_messages.values(1).struct_value().fields().at("operation").string_value());
-  EXPECT_EQ("db2.test2",
-            multiple_messages.values(1).struct_value().fields().at("resource").string_value());
+  EXPECT_TRUE(metadata.fields().find("db1.test1") != metadata.fields().end());
```
Comment (dio) [inline test/extensions/filters/network/mongo_proxy/proxy_test.cc]: I agree this is nicer.

File: source/extensions/filters/network/mongo_proxy/proxy.cc [hunk 2]
```diff
@@ -81,11 +81,9 @@ void ProxyFilter::setDynamicMetadata(std::string operation, std::string resource
             .dynamicMetadata()
             .mutable_filter_metadata())[NetworkFilterNames::get().MongoProxy]);
   auto& fields = *metadata.mutable_fields();
-  auto& list = *fields[DynamicMetadataKeysSingleton::get().MessagesField].mutable_list_value();
-  auto& message = *list.add_values()->mutable_struct_value()->mutable_fields();
-
-  message[DynamicMetadataKeysSingleton::get().OperationField].set_string_value(operation);
-  message[DynamicMetadataKeysSingleton::get().ResourceField].set_string_value(resource);
+  // TODO(rshriram): reverse the resource string (table.db)
```
Comment (dio) [inline source/extensions/filters/network/mongo_proxy/proxy.cc]: Super nit, add period at the end of the sentence (and also the other TODOs/comments).

**Comment by rshriram** [comment]
@dio the PR that implemented this was merged yesterday :). So users have not seen this stuff yet. So version history doesn't exist..

File: docs/root/configuration/network_filters/mongo_proxy_filter.rst [hunk 3]
```diff
@@ -182,11 +182,13 @@ Dynamic Metadata
...
+a list of operations performed on the collection.
 
 .. csv-table::
   :header: Name, Type, Description
   :widths: 1, 1, 2
 
-  operation, string, The operation to be executed.
-  resource, string, The resource name in *db.collection* format on which the operation is to be executed.
+  key, string, The resource name in *db.collection* format.
+  value, array, A list of strings representing the operations executed on the resource (insert/update/query/delete).
```
Comment (mattklein123) [inline docs/root/configuration/network_filters/mongo_proxy_filter.rst]: Per our offline convo, FYI, in order for this to be actually useful you are going to have to log/store query commands.

**Pull request merged**